apps/prairielearn/src/lib/editors.ts

```diff
@@ -1162,28 +1162,49 @@ export class QuestionModifyEditor extends Editor {
 }
 
 export class QuestionDeleteEditor extends Editor {
-  private question: Question;
+  private questions: Question[];
 
-  constructor(params: BaseEditorOptions) {
+  constructor(params: BaseEditorOptions & { questionBatch?: Question[] }) {
     super(params);
 
-    this.question = params.locals.question;
-    this.description = `Delete question ${this.question.qid}`;
+    if (params.questionBatch !== undefined) {
+      this.questions = params.questionBatch;
+    } else {
+      this.questions = [params.locals.question];
+    }
+
+    this.description =
+      this.questions.length === 1
+        ? `Delete question ${this.questions[0].qid}`
+        : `Delete questions ${this.questions.map((x) => x.qid).join(', ')}`;
   }
 
   async write() {
-    assert(this.question.qid, 'question.qid is required');
+    this.questions.forEach((question) => assert(question.qid, 'question.qid is required'));
 
     debug('QuestionDeleteEditor: write()');
-    await fs.remove(path.join(this.course.path, 'questions', this.question.qid));
-    await this.removeEmptyPrecedingSubfolders(
-      path.join(this.course.path, 'questions'),
-      this.question.qid,
-    );
+
+    await this.questions.forEach(async (question) => {
```

Comment:
This does not at all do what you want; `forEach` doesn't return anything. You can use a normal `for...of` loop instead (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of).

Reply:
Wasn't using the return-behavior but for ... of is cleaner; done in 253c7cd.